swift_browser_ui_frontend/vite.config.js

```diff
@@ -0,0 +1,172 @@
+
+import { defineConfig } from "vite";
+import { createVuePlugin as vue } from "vite-plugin-vue2";
+
+import fs from "node:fs";
+import path from "node:path";
+
+const TLS_ENABLED = process.env.VITE_TLS === "True";
+const http_mode = TLS_ENABLED ? "https" : "http";
+// paths should be absolute or relative to vite.config.js
+const TLS_CERT_PATH = process.env.VITE_TLS_CERT || undefined;
+const TLS_KEY_PATH = process.env.VITE_TLS_KEY || undefined;
+
+let https = null;
+if (TLS_ENABLED) {
+  console.log("vite dev serve will expect https");
+  https = {
+    key:  fs.readFileSync(TLS_KEY_PATH),
+    cert: fs.readFileSync(TLS_CERT_PATH),
+  };
+}
+
+const proxyTo = {
+  target: `${http_mode}://${process.env.BACKEND_HOST || "localhost"}:${process.env.BACKEND_PORT || "8080"}`,
+  changeOrigin: true,
+  secure: false,  // Won't check certificates
+};
+
+const oidcEnabled = process.env.OIDC_ENABLED === "True";
+const root = path.resolve(__dirname, "src");
+let pages = {
+  "index":         path.resolve(root, "index.html"),
+  "select":        path.resolve(root, "select.html"),
+  "badrequest":    path.resolve(root, "400.html"),
+  "unauth":        path.resolve(root, "401.html"),
+  "forbid":        path.resolve(root, "403.html"),
+  "notfound":      path.resolve(root, "404.html"),
+  "uidown":        path.resolve(root, "503.html"),
+  "browse":        path.resolve(root, "browse.html"),
+  "loginpassword": path.resolve(root, "loginpassword.html"),
+  "login":         path.resolve(root, "login.html"),
+  "login2step":    path.resolve(root, "login2step.html"),
+};
+if (oidcEnabled) {
+  pages["index"] = path.resolve(root, "index_oidc.html");
+}
+let proxy = {
+  "/static/assets":       proxyTo,
+  "/api":                 proxyTo,
+  "/discover":            proxyTo,
+  "/libupload":           proxyTo,
+  "/login/oidc":          proxyTo,
+  "/login/oidc_front":    proxyTo,
+  "/login/oidc-redirect": proxyTo,
+  "/login/credentials":   proxyTo,
+  "/login/return":        proxyTo,
+  "/login/rescope":       proxyTo,
+  "/upload":              proxyTo,
+  "/enupload":            proxyTo,
+  "/download":            proxyTo,
+  "/sign":                proxyTo,
+  "/replicate":           proxyTo,
+  "/token":               proxyTo,
+  "/ws": {
+    target: `ws${process.env.SWIFT_UI_SECURE_WEBSOCKET}://${process.env.SWIFT_UI_TLS_HOST}:${process.env.SWIFT_UI_TLS_PORT}/ws`,
+    ws: true,
+  },
```

Comment:
What exactly is this meant to proxy? In the old implementation this approach was chosen, to get the websocket URLs to be correct when connecting to Nginx, in this version it just seems to create a circular proxy at first glance. Vite can't connect to the ws when I'm testing this in my own environment, and I can't figure out what I'm missing 😅 

![Screenshot from 2023-03-16 12-01-38](https://user-images.githubusercontent.com/24293337/225583826-9846b880-e149-4841-bc34-326f2bf65d7c.png)

Reply:
The Vite dev server doesn't have the same kind of configuration as the webpack-dev-server. So I moved the websockets config to the proxy configuration.

https://vitejs.dev/config/server-options.html

Reply:
Changed it a bit to allow using the websocket, also fixed the `publicDir` to allow using `libupload.js` and `libupload.wasm` without proxying to backend.

https://github.com/CSCfi/swift-browser-ui/pull/1035/files

Reply:
thanks